pkg/check/table_structure.go

```diff
@@ -161,8 +159,18 @@ func (c *TablesChecker) Name() string {
 }
 
 func (c *TablesChecker) checkCreateSQL(statement string) []*incompatibilityOption {
+	ansiQuotes, err := dbutil.HasAnsiQuotesMode(c.db)
+	if err != nil {
+		return []*incompatibilityOption{
+			{
+				state:      StateFailure,
+				errMessage: err.Error(),
+			},
+		}
+	}
+
 	sqlMode := ""
-	if c.enableANSIQuotes {
+	if ansiQuotes {
 		sqlMode = "ANSI_QUOTES"
 	}
 	parser2, err := dbutil.GetParser(sqlMode)
```

Comment:
How about combine them into one function?

Reply:
done in https://github.com/pingcap/tidb-tools/pull/381/commits/3dccc006997826ff0344dc233afda8776d223410